src/VisualStudio/Core/Def/Implementation/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.Factory.cs

```diff
@@ -32,7 +29,7 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
             if (_singleton == null)
             {
                 var temporaryStorage = workspaceServices.GetService<ITemporaryStorageService>();
-                Interlocked.CompareExchange(ref _singleton, new VisualStudioMetadataReferenceManager(workspaceServices.Workspace, _serviceProvider, temporaryStorage), null);
+                Interlocked.CompareExchange(ref _singleton, new VisualStudioMetadataReferenceManager(_serviceProvider, temporaryStorage), null);
```

Comment:
❗ This incorrectly provides a `VisualStudioMetadataReferenceManager` using the `ITemporaryStorageService` from an unrelated workspace. The cache field needs to be removed.

Reply:
This bug isn't introduced by this pull request, but if you aren't fixing it here please file a bug for the 17.1 release.